### ISSUE TITLE
May I help ? 

### ISSUE THREAD

**Issue by samchery** [OpenHogwarts/hogwarts#52]
I'm not a Unity-graph-modeler or anything else. Just a programmer, usually web-backend-php-db like sql. But, I'm in love with this universe, i know it by heart, I may help with the scenario/game play/quest if you want :) Also, C# shouldn be to difficult for me to understand, just need to read a bit about the langage's rules, but coming from php, i already know the big concept about oop and dev so ... may i help with that :)
I learned Spanish at school, old days... I still can understand it (with google's help for some words hahaha), maybe able to translate some of your text to french/english :)
so let me know, i m going to look around a bit in the project tonight, but would be so happy to help you in this wonderful adventure ! Your project is so amazing, it could be so great to play with for real.

**Comment by OpenHogwarts** [comment]
Hi Sam!

The game should already be available in English :S  (https://github.com/OpenHogwarts/hogwarts/blob/master/Assets/Resources/i18n/English.json)
¿Do you see it in Spanish? If yes, could you debug it?
Here is the manager class: https://github.com/OpenHogwarts/hogwarts/blob/master/Assets/Plugins/LanguageManager.cs

If you're coming from php you will have it easy to understand/write c# :)
- For debugging in unity use  `Debug.Log("say hi!")`

Our biggest problem is that we don't have "tons of" monsters/npcs :(
We can't have a good gameplay because of this. There can't be quests if there aren't monsters to kill or npcs to talk. Map areas also look empty without animals moving around.

Right now, this is the ToDo:
- [Animator/Modeling] We need more creatures/npcs. The models can be obtained from multiple places like https://www.youtube.com/watch?v=UjK-ZQ7By4U or Google warehouse (example: https://3dwarehouse.sketchup.com/model/8f9e8397-dd10-482a-81a4-457dbd0768a9/Hagrid), but they don't have "skeleton", so can't be animated. 
We haven't yet found a contributor that knows how to properly add them (it can be done with Blender, but 0 knowledge of using it).
- [Coding] There is no "dead/respawn screen". If i had to do it myself, i would do one of this options, but maybe you come on another solution:
+ When dead, a death screen appears and only has a button that lets you respawn. It will bring you back to hogwarts, you will lose some coins or items/xp and half of health bar.
+ When dead, you will appear in the nearest graveyard as ghost and you will have to go back to where your character died to respawn. When respawned you will have half of health bar.
- [Coding] There is no skill tree. RPGs usually have a skill tree. In hp world you should be able to learn all spells that appear in the books.
- [Coding] Combat system is quite sucky/boring. Dunno how to make it cool.

**Comment by samchery** [comment]
Oh ! Mine is in spanish hahaha I will look to change langage and add some french ... But I may have some pbm because I can't change the point of view of the camara, I mean, looking more from the top ...

As I said, I can't modeling neither Animate. But, maybe https://www.mixamo.com/#/ is a solution to animate the available model with pre-formated movement ? Do you know this ? Looks like it's easy...

I will look at the skill tree if you want to develop it :) Do you have special zone / team to discuss about how the game should works ? :)

**Comment by OpenHogwarts** [comment]
That's strange, are you trying the game from the build or in unity?
Mixamo got purchased by Adobe, now barely works (tried, ignores uploaded models :()
Well, there is no team at all, just collaborators that make changes when they can. There was a discord but i think no one (including myself) join it. So noe, there is no other place to discuss than github issues :o

**Comment by samchery** [comment]
Arg... I opened it in unity (play en MainMenu), what is from the build? Some words are in english but not all of them actually... And some are inverted as in mirror.
And for Mixamo, bad luck. It takes long to animate properly with blender... 
Ok so I will try to look at what I can do and then post here :) You are currently working on what ?

**Comment by OpenHogwarts** [comment]
It should be ok, as i don't know how much you know about Unity, let me ask for the basics:
![https://i.imgur.com/geRMYgR.png](https://i.imgur.com/geRMYgR.png)

1. Maybe you're checking "Scene" tab which may have 2D toggle set?
2. Make sure you're on Game tab and click Play button

¿Those inverted texts are character names?

I'm unsuccessfully attempting to place skeletons on biped models & searching for more models/scenarios

**Comment by OpenHogwarts** [comment]
I don't think so :S

I can do teamviewer to try to find what the heck is happening if you want.

**Comment by samchery** [comment]
I will have a look may have some errors... but if i cant figure it out, maybe yes, we can plan that.

**Comment by samchery** [comment]
I had 2017.4.11.f, I just downloaded the 4.1 version (can't find the 4.0), still didn't work ... Will investigate more tomorrow night...

**Comment by OpenHogwarts** [comment]
Wow, that's strange, unity didn't update https://github.com/OpenHogwarts/hogwarts/blob/master/ProjectSettings/ProjectVersion.txt
But the lastest checked build is 2018.1.6f (https://github.com/OpenHogwarts/hogwarts/commits/master)

**Comment by samchery** [comment]
still not working ... Do you have as well a lot of warning ? I found an error (cinema 4D not found), i m trying to fix that.

**Comment by OpenHogwarts** [comment]
Cinema 4d should not be required for anything :s
Print those warnings

**Comment by OpenHogwarts** [comment]
Warnings are warnings, they just say that some of the models have to many vertices, (again, there are no modelers so we can't do anything) but that won't make the game explode.

The cinema error isn't also important, just lets you know that unity won't handle .cd4 files, which is ok, as we are not using them ingame. You can make an easy test, look for *.c4d files in the project and delete them. The "error" will disappear and nothing will change ingame. They are just source model files :)

**Comment by finchen** [comment]
I managed to import mixamo model and animations. I need to tidy it up and I will commit in a branch

**Comment by finchen** [comment]
This is the process to rig and import sketchup model with mixamo.
I am planning to import a few but I will mainly work on the NPC AI (and try things) and perhaps UI/Quest stuff. So feel free to start to import more.
Note: This is only working for humanoid. Ideally the wand might need to be removed for some characters.

1. Download a sketchup model .skp _eg. Draco_
https://3dwarehouse.sketchup.com/collection/e392d7b8-4004-4b22-839b-4a9f9c86af02/Harry-Potter?sortBy=modifyTime%20DESC&searchTab=model

2. Open in sketchup pro (trial work for 30 days) and export to 3D model .obj
You will get few files
- draco.obj
- draco.mtl
- draco/ 

3. zip these files. draco.zip
4. Upload the zip in mixamo
5. Rotate then Rig
6. Choose Idle animation (but it does not really matter)
7. Download 
- fbx for unity
- with skin
8. Import the .fbx into unity assets (drag&drop .fbx tino Assets\Models\Characters)
9. Click on model in unity assets, then **Inspectors** then **Materials** tab. 
**Extract Textures** to a _new folder_ that you create (eg. draco)
10. **Model** tab. Change scale factor to around _0.005_ (might differ)
11. Ready to be used with any mixamo animation

![image](https://user-images.githubusercontent.com/1032201/61509179-7be96400-aa41-11e9-8948-3ea4dfcfa11f.png)

**Comment by finchen** [comment]
I have imported draco and professor quirel, created their prefab and placed them near the player spawn.
Animation must not be "Apply root motion" to work with current NPC class.
I will improve before to create a pull request
https://github.com/finchen/hogwarts/tree/npc-mixamo

**Comment by OpenHogwarts** [comment]
Cool, i think npc script didn't work with animator because the spider only had animations with the old format.

**Comment by finchen** [comment]
I made it work so it is using the animator from the model if any, and fallback on old animator.

**Comment by OpenHogwarts** [comment]
@finchen this may be useful:
https://www.deviantart.com/kittyinhiding/art/Harry-Potter-Wizards-Unite-803313402

**Comment by finchen** [comment]
They are looking good but not sure about using assets from another game.
We could use basic model in the game and let the user download and install these type of model..

**Comment by OpenHogwarts** [comment]
Yep finchen im still far from a good internet connection so i won't be able to merge anything these days. About using models from other games, we're already using them since hogwarts model is from an original hp game, and by simply naming hogwarts or hp we're already making a copyright infraction "because laws" so I wouldn't worry about using even more. They already have the right to take down the project whenever they want.
Switching to to the PR you made. You added a debug bar to test the health, there is a test scene that you can also join which is way lighter than the castle, so it loads faster and it has some red cubes that give or take health to exactly test that ;)

**Comment by finchen** [comment]
ok good to know. it is for educational purpose anyway.
I might use their model then. I am keen to learn a bit about kinematic (eg. animate a dragon) but I don't want to spend too much time on that. I prefer coding.

**Comment by finchen** [comment]
@OpenHogwarts  how to you feel about adding a navmesh? I am keen to create wandering NPC (walking around and doing stuff) or NPC (rat, dragon) that the player will try to catch

**Comment by OpenHogwarts** [comment]
We integrated a waypoint system to avoid having much lag with thousands of npcs if i'm not wrong (https://www.youtube.com/watch?v=GC8x9vIplEs) that way they also behave like most MMOs where you can predict their routine. Maybe we should migrate to unity ECS & job system, that is able to having thousands of npcs on realtime (https://youtu.be/GEuT5-oCu_I?t=1035) :s

**Comment by finchen** [comment]
Ok thanks. I ll have a look

About Harry-Potter-Wizards-Unite-803313402. so far
- humanoid. can import. need to change scale to 1.5 on prefab. There is no animation.
They are rigged but mixamo animations don't work on them. Not sure why.
I will try to later to upload the model to mixamo and see if it can autorig it.
- non humanoid
They are rigged but I still don't know how to animate them. I have to find the right/easy way to do it. 
- others like books etc. It will be easy to import even if they are not animated to start with.

**Comment by OpenHogwarts** [comment]
uh, so strange, i will try to take a look at them this weekend and post any result. Can you see the bones? Maybe misses root animation check?
